prime-router/docs/onboarding-users/transport/rest.md

```diff
@@ -0,0 +1,165 @@
+    <li>Reading credential from Azure Vault, RESTTransport uses its Type to determine the authentication type (i.e., UserPass or UserApiKey).  Next, it uses the obtained key to authenticates with STLT and sends the message payload.</li>
+</ol>
+
+## 2.) How to prepare a key for authentication
+For authentication, you need to do both step a and step b.
+<ol type="a">
+<li>Generate the "Credential in JSON format" for authentication</li>
+
+Currently, RESTTransport uses the one of the three options: i) UserPass, ii) UserApiKey with JKS, or iii) UserApiKey with two-legged credential type to authenticate and obtain Bearer token from STLT. User can use **primeCLI** command with credential-create option to generate the "Credential in JSON format" as given below.
+
+- With STLT's credential username and password given to us by STLT, user needs to run the following command to generate the UserPass credential type object:
+
+        Command:
+            ./prime credential-create --type UserPass --user <username> --pass <password>
+        
+        Output: 
+            Credential in JSON format: 
+            ===
+            {"@type":"UserPass","user":"<username>","pass":"<password>"}
+            ===
+
+- With STLT's credential username and API Key given to us by STLT, user needs to run the following command to generate the UserApiKey credential type object:
+	
+          Command:
+              ./prime credential-create --type UserApiKey --apikey-user <username> --apikey <API Key>
+        
+          Output: 
+              Credential in JSON format: 
+              ===
+              {"@type":"UserApiKey","user":"<username>","key":"<API Key>"}
+              ===
+
+- With STLT's credential jks-Username, jks-File-Password, and jks-key-file given to us by STLT, user needs to run the following command to generate the UserJks credential type object:
+	
+        Command:
+            ./prime credential-create --type UserJks --jks-use <JKS Username> --jks-file-pass <JKS File Password> --jks-file <PATH to JKS File>
+            
+        Output: 
+            Credential in JSON format: 
+            ===
+            {"@type":"UserJks","user":"<JKS Username>", "jks":"<JKS Key>"}
+            ===
+
+<li>Create Azure Key Vault Secret for STLT</li>
+
+With CDC SU password, login into the **Microsoft Azure** and navigate to the **pdhprod-clientconfig** and select **Secrets** from the menu on the left.  You should see the following screenshot.
+
+![](./images/rest-pdhprod-clientconfig-secrets.png)
+
+Next refer to the above image, click on the  [+ Generate/Import] as the red arrow is pointing to.  You should see the following screenshot. Please follow the instruction given in the image.
+
+![](./images/rest-create-a-secrete-key-screenshot.png)
+
+</ol>
+
+## 3.) How to configure the receiver's REST transport
+The receiver's RESTTransport includes the following fields:
+
```

Comment:
How is the secret we create in Azure used or referenced when we set-up the receiver's transport? Is it retrieved by looking at the receiver's name?

Reply:
It get referenced by OrganizationName, and Receiver.name (e.g., NY-PHD--ELR).